### PR TITLE
Ankit | Test | TEST - Connect to bank button is not coming on onboarding page where gocardless is integrated

### PR DESCRIPTION
**Pull request by ankitdubeyoffice**
https://app.clickup.com/t/86cwj2dhb

### CODE REVIEW

**Review by nisha-walkover — APPROVED** [review]

File: apps/web-giddh/src/app/onboarding/onboarding.component.html [hunk 1]
```diff
@@ -38,7 +38,7 @@ <h1 class="border-btm">{{ localeData?.try_things }}</h1>
                                 {{ localeData?.create_bank_account }}
                             </a>
                         </mat-list-item>
-                        <mat-list-item role="listitem" class="mr-b1 px-0" *ngIf="(companyCountry !== 'India') && isPlaidSupportedCountry">
+                        <mat-list-item role="listitem" class="mr-b1 px-0" *ngIf="isGocardlessSupportedCountry || (companyCountry !== 'India') && isPlaidSupportedCountry">
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/onboarding/onboarding.component.html]:
```suggestion
                        <mat-list-item role="listitem" class="mr-b1 px-0" *ngIf="(companyCountry !== 'India') && (isPlaidSupportedCountry || isGocardlessSupportedCountry)">
```

File: apps/web-giddh/src/app/onboarding/onboarding.component.ts [hunk 2]
```diff
@@ -49,14 +51,17 @@ export class OnboardingComponent implements OnInit, AfterViewInit, OnDestroy {
     private createAccountIsSuccess$: Observable<boolean>;
     /** Holds true if current company country is plaid supported country */
     public isPlaidSupportedCountry: boolean;
+    /** Holds true if current company country is gocardless supported country */
+    public isGocardlessSupportedCountry: boolean = false;
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/onboarding/onboarding.component.ts]:
```suggestion
    public isGoCardlessSupportedCountry: boolean = false;
```

File: apps/web-giddh/src/app/onboarding/onboarding.component.ts [hunk 3]
```diff
@@ -49,14 +51,17 @@ export class OnboardingComponent implements OnInit, AfterViewInit, OnDestroy {
...
     /** Stores the voucher API version of current company */
     public voucherApiVersion: 1 | 2 = 2;
 
     constructor(
         private _router: Router, private _generalService: GeneralService,
         private store: Store<AppState>,
         private settingsProfileActions: SettingsProfileActions,
-        private generalActions: GeneralActions
+        private generalActions: GeneralActions,
+        private componenet: OnboardingComponentStore
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/onboarding/onboarding.component.ts]:
```suggestion
        private componentStore: OnboardingComponentStore
```

File: apps/web-giddh/src/app/onboarding/onboarding.component.ts [hunk 4]
```diff
@@ -72,6 +77,12 @@ export class OnboardingComponent implements OnInit, AfterViewInit, OnDestroy {
             }
         });
 
+        this.componenet.companyProfile$.pipe(takeUntil(this.destroyed$)).subscribe((profile) => {
+             if (profile && profile.countryV2 && profile.countryV2.alpha2CountryCode) {
+                 this.isGocardlessSupportedCountry = this._generalService.checkCompanySupportGocardless(profile.countryV2.alpha2CountryCode);
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/onboarding/onboarding.component.ts]:
```suggestion
                 this.isGoCardlessSupportedCountry = this._generalService.checkCompanySupportGoCardless(profile.countryV2.alpha2CountryCode);
```

**Review by dvCodeWorld — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/onboarding/onboarding.component.html (hunk 1 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/onboarding/onboarding.component.html]: Done

File: apps/web-giddh/src/app/onboarding/onboarding.component.ts (hunk 2 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/onboarding/onboarding.component.ts]: Done

File: apps/web-giddh/src/app/onboarding/onboarding.component.ts (hunk 3 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/onboarding/onboarding.component.ts]: Done

File: apps/web-giddh/src/app/onboarding/onboarding.component.ts (hunk 4 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/onboarding/onboarding.component.ts]: Done

**Review by dvCodeWorld — APPROVED** [review]

**Review by dvCodeWorld — APPROVED** [review]

**Pull request merged**